WooCommerce/Classes/ViewModels/Order Details/OrderDetailsDataSource.swift

```diff
@@ -24,12 +24,24 @@ final class OrderDetailsDataSource: NSObject {
     ///
     private(set) var sections = [Section]()
 
-    /// Is this order processing?
+    /// Is this order complete?
     ///
-    private var isProcessingPayment: Bool {
+    private var isCompletedStatus: Bool {
+        return order.status == OrderStatusEnum.completed
+    }
+
+    /// Is this order processing? Payment received (paid). The order is awaiting fulfillment.
+    ///
+    private var isProcessingStatus: Bool {
```

Comment:
Definitely, made the change at 449bdf49e73e39d8e4ba0e777f69502ef6bfad1e

This uncovers a bit of an edge case: `needsProcessing` returns false when the Order is made of Digital Downloadable products, as these do not need processing, however, we allow these orders to be in processing status regardless which breaks all the `Mark as Completed` logic. For the time being, I went ahead with the following workaround:

```
    /// Does this order need processing? Payment received (paid). The order is awaiting fulfillment.
    /// All product orders require processing, except those that only contain products which are both Virtual and Downloadable.
    ///
    private var needsProcessing: Bool {
        // We check both the order.status and order.needsProcessing as a workaround to allow Digital Downloadable Orders to be in processing status
        ( order.status == OrderStatusEnum.processing || order.needsProcessing )
    }
```

What do you think?

Reply:
🤔 My instinct is that if we want to check if an order needs processing, we should trust that `needsProcessing` property without having to add other checks.

I believe the only scenario in which an order with only downloadable products will be in the `processing` status is if the status is edited manually. (In the usual fulfillment workflow, once that order is paid for it is marked complete, skipping the `processing` status.) Given that, it seems reasonable for the merchant to mark it complete by editing the status manually again.

WDYT?

Reply:
Also, don't forget to update the `Order` entity in networking to add a fallback for the `needsProcessing` property. That's needed on stores with < WC 6.6, since they won't have that in the API response.